From 430a4b50f2f7e5f21c7c2f81849588c575018501 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jaime=20Caama=C3=B1o=20Ruiz?= <jcaamano@suse.com>
Date: Mon, 21 Sep 2020 14:50:13 +0200
Subject: [PATCH] SLE15 SP3 compatibility patch for kni

  Add patch to resolve build error reported in Staging:E project 
  where a kernel backport (jsc#SLE-13536) introduced backward
  incompatible changes to the API that break kni module build
  * 0001-SLE15-SP3-compatibility-patch-for-kni.patch

Updated 2023-02-26 for version 22.11

diff --git a/kernel/linux/kni/compat.h b/kernel/linux/kni/compat.h
index 3a86d12bbc..66d1afd944 100644
--- a/kernel/linux/kni/compat.h
+++ b/kernel/linux/kni/compat.h
@@ -14,7 +14,10 @@
 #define SLE_VERSION(a, b, c) KERNEL_VERSION(a, b, c)
 #endif
 #ifdef CONFIG_SUSE_KERNEL
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 4, 57))
+#if (defined(CONFIG_SUSE_VERSION) && CONFIG_SUSE_VERSION == 15 && \
+     defined(CONFIG_SUSE_PATCHLEVEL) && CONFIG_SUSE_PATCHLEVEL == 3)
+#define SLE_VERSION_CODE SLE_VERSION(15, 3, 0)
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 4, 57))
 /* SLES12SP3 is at least 4.4.57+ based */
 #define SLE_VERSION_CODE SLE_VERSION(12, 3, 0)
 #elif (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 12, 28))
@@ -132,6 +135,7 @@
 #endif
 
 #if KERNEL_VERSION(5, 6, 0) <= LINUX_VERSION_CODE || \
+	(SLE_VERSION_CODE && SLE_VERSION_CODE == SLE_VERSION(15, 3, 0)) || \
 	(defined(RHEL_RELEASE_CODE) && \
 	 RHEL_RELEASE_VERSION(8, 3) <= RHEL_RELEASE_CODE) || \
 	 (defined(CONFIG_SUSE_KERNEL) && defined(HAVE_ARG_TX_QUEUE))
